codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -424,8 +424,7 @@ let handlePartitionQueryResponse = (
               blockNumberThreshold: lastBlockScannedData.blockNumber -
               updatedChainFetcher.chainConfig.confirmedBlockThreshold,
               blockTimestampThreshold: chainManager
-              ->ChainManager.getEarliestMultiChainTimestampInThreshold
-              ->Option.getWithDefault(0),
```

Comment:
Ah so this was the bug beforehand?

Reply:
Yep

Reply:
Actually, the `LastBlockScannedHashes` would operate only with blocks in reorg threshold (<200), but it only prunes blocks on response, so if the indexer restarts and the first query detects a rollback, then there'd be 15k+ calls to hypersync.